src/main.rs

```diff
@@ -745,21 +745,54 @@ impl TypeDarwin {
     }
 }
 
-//     def count_argument_dashes(self, line):
-//         # Determine how many dashes the line has using the following regex hack
-//         # Look for the start of a line, followed by a dot, then a sequence of
-//         # one or more dashes ('Fl')
-//         result = 0
-//         if line.startswith('.'):
-//             line = line[4:]
-//             while line.startswith('Fl '):
-//                 result = result + 1
-//                 line = line[3:]
-//         return result
+#[test]
+fn test_replace_all() {
+    let (string, num) = replace_all("");
+    assert_eq!(string, "");
+    assert_eq!(num, 0);
+
+    let (string, num) = replace_all(".xyzppp");
+    assert_eq!(string, "ppp");
+    assert_eq!(num, 0);
+
+    let (string, num) = replace_all("Fl jkl");
+    assert_eq!(string, "Fl jkl");
+    assert_eq!(num, 0);
+
+    let (string, num) = replace_all(".xxxFl jkl");
+    assert_eq!(string, "jkl");
+    assert_eq!(num, 1);
+
+    let (string, num) = replace_all(".Fl Fl Fl jkl");
+    assert_eq!(string, "jkl");
+    assert_eq!(num, 2);
+}
+
+use std::borrow::Cow;
+fn replace_all(line: &str) -> (Cow<str>, u32) {
+    let mut result = 0;
+    (
+        regex!(r"^(?:\....)((?:[F][l]\s)*)").replace(&line, |captures: &regex::Captures| {
+            result = (captures[1].len() / 3) as u32; // Divide by 3 since there are 3 bytes per `Fl\s` pattern
+            ""
+        }),
+        result,
+    )
+}
+
+#[test]
+fn test_TypeDarwin_count_argument_dashes() {
+    assert!(TypeDarwin::count_argument_dashes(".Fl Fl xx") == 1);
```

Comment:
Would be a little more idiomatic to do `assert_eq!(X, Y)` rather than `assert!(X == Y)` for this and the following lines.